src/chains/ethereum/ethereum/tests/forking/fork.test.ts

```diff
@@ -0,0 +1,130 @@
+import { EthereumOptionsConfig } from "@ganache/ethereum-options";
+import { Fork } from "../../src/forking/fork";
+import { KNOWN_CHAINIDS, Quantity } from "@ganache/utils";
+import { Common } from "@ethereumjs/common/dist/common";
+import ganache from "../../../../../packages/core";
+import Server from "../../../../../packages/core/lib/src/server";
+import assert from "assert";
+import { logging } from "./helpers";
+
+describe("Fork", () => {
+  const port = 9999;
+  const networkId = 1;
+  const accounts = [];
+  const forkOptions = {
+    fork: {
+      url: `http://localhost:${port}`
+    },
+    logging
+  };
+
+  let remoteServer: Server;
+  let fork: Fork;
+
+  before(async () => {
+    remoteServer = ganache.server({
+      wallet: { deterministic: true },
+      chain: { networkId: networkId },
+      logging
+    });
+    await remoteServer.listen(port);
+  });
+
+  beforeEach(async () => {
+    const providerOptions = EthereumOptionsConfig.normalize(forkOptions);
+    fork = new Fork(providerOptions, accounts);
+    await fork.initialize();
+  });
+
+  afterEach(async () => {
+    await fork.close();
+  });
+
+  after(async () => {
+    await remoteServer.close();
+  });
+
+  describe("getCommonForBlock()", () => {
+    it("should return a Common for known chainIds", () => {
+      KNOWN_CHAINIDS.forEach(chainId => {
+        if (chainId === 42) {
+          // Skip kovan, because it is no longer supported by ethereumjs. To be
+          // removed in https://github.com/trufflesuite/ganache/issues/4461
+        } else {
+          assert.doesNotThrow(() => {
+            const parentCommon = new Common({ chain: chainId });
+
+            fork.getCommonForBlock(parentCommon, {
+              number: 0n,
+              timestamp: 0n
+            });
+          });
+        }
+      });
+    });
+
+    it("should resolve the correct hardfork based on block number for known chainId", () => {
+      const mainnet = 1;
+      const mergeBlocknumber = 15537394n;
+
+      // ensure that the fork blockNumber is after the merge blockNumber
+      fork.blockNumber = Quantity.from(mergeBlocknumber + 100n);
+      fork.chainId = mainnet;
+
+      const parentCommon = new Common({ chain: mainnet });
+      const blocknumberToHardfork: [bigint, string][] = [
+        [mergeBlocknumber - 1n, "grayGlacier"],
+        [mergeBlocknumber, "merge"],
+        [mergeBlocknumber + 1n, "merge"]
+      ];
+
+      blocknumberToHardfork.forEach(([number, expectedHardfork]) => {
+        const common = fork.getCommonForBlock(parentCommon, {
+          number, // the block at which paris is scheduled
```

Comment:
Whats this about paris?

Reply:
apologies, the base branch was wrong here - as it depends on #4455 where I've fixed this, along with a minor change to the above comment https://github.com/trufflesuite/ganache/pull/4455/files/cb95798713916bda17494b7129aa8eb99220a786..163c987b7d9dbdf24eee0cc5569af085cf318989